internal/pkg/helper/tokens.go

```diff
@@ -0,0 +1,93 @@
+package helper
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"github.com/assetto-corsa-web/accweb/internal/pkg/cfg"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+func GenerateTokenKeysIfNotPresent(config *cfg.Config) {
+	// Based on https://stackoverflow.com/questions/64104586/use-golang-to-get-rsa-key-the-same-way-openssl-genrsa
+	bitSize := 4096
+
+	logrus.Info("accweb: checking for secrets...")
+
+	// make sure files don't already exist
+	if Exists(config.Auth.PublicKeyPath) && Exists(config.Auth.PrivateKeyPath) {
+		logrus.WithField("file", config.Auth.PublicKeyPath).Info("Public/private keys already exists, not attempting regeneration")
+		return
+	} else if Exists(config.Auth.PublicKeyPath) {
```

Comment:
```suggestion
	} else if !Exists(config.Auth.PublicKeyPath) {
```

Must be if NOT exists

Reply:
No, these checks make sure we don't overwrite existing token files if they already exist, so we check if they exist and then return.